listenbrainz/webserver/views/atom.py

```diff
@@ -0,0 +1,242 @@
+    :statuscode 200: The feed was successfully generated.
+    :statuscode 400: Bad request.
+    :statuscode 404: The user does not exist.
+    :resheader Content-Type: *application/atom+xml*
+    """
+    user = db_user.get_by_mb_id(db_conn, user_name)
+    if user is None:
+        return Response(status=404)
+
+    # Get and validate interval
+    interval = request.args.get("interval", 60)
+    if interval:
+        try:
+            interval = int(interval)
+        except ValueError:
+            return Response(status=400)
+
+    if interval < 1:
+        return Response(status=400)
+
+    # Construct UTC timestamp for interval
+    from_ts = datetime.now() - timedelta(minutes=interval)
+
+    listens, _, _ = timescale_connection._ts.fetch_listens(user, from_ts=from_ts)
+
+    fg = FeedGenerator()
+    # TODO: use {base_url} instead of hardcoded url
+    fg.id(f"https://listenbrainz.org/user/{user_name}")
+    fg.title(f"Listens for {user_name} - ListenBrainz")
+    fg.author({"name": "ListenBrainz"})
+    fg.link(href=f"https://listenbrainz.org/user/{user_name}", rel="alternate")
+    fg.link(
+        href=f"https://listenbrainz.org/syndication-feed/user/{user_name}/listens",
+        rel="self",
+    )
+    fg.logo("https://listenbrainz.org/static/img/listenbrainz_logo_icon.svg")
+    fg.language("en")
+
+    # newer listen comes first
+    for listen in reversed(listens):
+        fe = fg.add_entry()
+        # according to spec, ids don't have to be deferencable.
+        fe.id(
+            f"https://listenbrainz.org/syndication-feed/user/{user_name}/listens/{listen.ts_since_epoch}/{listen.data['track_name']}"
+        )
+        fe.title(f"{listen.data['track_name']} - {listen.data['artist_name']}")
+
+        _content = render_template("atom/listens.html",
+            user_name=listen.user_name,
+            track_name=listen.data["track_name"],
+            recording_mbid=listen.data["additional_info"].get("recording_mbid"),
+            artist_name=listen.data["artist_name"],
+            artist_mbid=listen.data['additional_info'].get('artist_mbid'),
+            time=listen.timestamp.strftime("%Y-%m-%d %H:%M:%S"),
+            release_mbid=listen.data["additional_info"].get("release_mbid"),
+            release_name=listen.data["release_name"],
+        )
+
+        fe.content(
+            content=_content,
+            type="html",
+        )
+        fe.published(listen.timestamp)
+        fe.updated(listen.timestamp)
+
+    atomfeed = fg.atom_str(pretty=True)
+
+    return Response(atomfeed, mimetype="application/atom+xml")
+
+
+@atom_bp.route("/fresh_releases", methods=["GET"])
+@crossdomain
+@ratelimit()
+def get_fresh_releases():
+    """
+    Get site-wide fresh releases.
+
+    :param days: The number of days of fresh releases to show. Max 90 days.
```

Comment:
For docstrings, it is better to avoid the constants and add a reference to DEFAULT_NUMBER_OF_FRESH_RELEASE_DAYS instead.

Reply:
Great!